Library/Homebrew/cmd/formulae.sh

```diff
@@ -13,9 +13,13 @@ homebrew-formulae() {
   if [[ -z "${HOMEBREW_NO_INSTALL_FROM_API}" &&
         -f "${HOMEBREW_CACHE}/api/formula_names.txt" ]]
   then
-    cat "${HOMEBREW_CACHE}/api/formula_names.txt"
+    {
+      cat "${HOMEBREW_CACHE}/api/formula_names.txt"
+      echo
+      homebrew-items '*\.rb' '.*Casks.*|.*/homebrew/homebrew-core/.*' 's|/Formula/|/|' '^homebrew/core'
+    } | sort -f
     echo
   else
-    homebrew-items '*\.rb' 'Casks' 's|/Formula/|/|' '^homebrew/core'
+    homebrew-items '*\.rb' '.*Casks.*' 's|/Formula/|/|' '^homebrew/core'
```

Comment:
```suggestion
    homebrew-items '*\.rb' '.*/Casks/.*' 's|/Formula/|/|' '^homebrew/core'
```

Nit: It might be nice to be a little more defensive here and with the regex above it too.

Reply:
Agreed.

Reply:
I'm not sure why but adding slashes breaks the filtering and no longer excludes casks.

Reply:
It seems to be because we're not filtering by file but directory instead and the final slash is omitted. The following works for me locally. If I remember correctly, we don't currently use nesting but it will probably get added to the core repos in the future for git performance reasons.

```suggestion
    homebrew-items '*\.rb' '.*/Casks(/.*|$)' 's|/Formula/|/|' '^homebrew/core'
```